Entry.java

```diff
@@ -96,7 +98,16 @@ public void selectAmount(){
      * Select a date when the purchase was made
      */
     public void selectDate(){
-        this.date = new Date();
+
+        this.date = new validDate();
+    }
+
+    /**
+     * Lets the user input a guilty level based on his purchase:
+     */
+    public void selectGuiltyLevel(){
+        System.out.println("Finally, please enter how guilty you feel about this purchase, on a scale from 1 to 10:");
+        this.guiltyLevel = userIn.nextInt();
```

Comment:
does thie need a try catch if they enter a non-integer value? or does nextInt() not read it if it isnt an int?

Reply:
Not sure, didn't think about that, will look into it today and update accordingly.

Reply:
Yea a try catch should be needed for Input mismatch exception and an if/else statement for checking that is positive and between 1 and 10 (or 0 and 10)